Source/Rhetos.Utilities/Legacy/Paths.cs

```diff
@@ -83,10 +83,14 @@ private static void AssertRhetosRootPathNotNull()
 
         private static void ValidateGeneratedFolder()
         {
-            if (NonNullRhetosAppOptions.AssetsFolder == null && NonNullBuildOptions.GeneratedAssetsFolder == null)
+            if (_appOptions == null && _buildOptions == null)
+                throw new FrameworkException($"Rhetos server is not initialized ({nameof(Paths)} class)." +
+                    $" Use {nameof(LegacyUtilities)}.{nameof(LegacyUtilities.Initialize)}() to initialize obsolete static utilities");
+
+            if (_appOptions?.AssetsFolder == null && _buildOptions?.GeneratedAssetsFolder == null)
                 throw new FrameworkException($@"One of the following value should be set. {nameof(RhetosAppOptions.AssetsFolder)} or {nameof(BuildOptions.GeneratedAssetsFolder)}");
 
-            if (NonNullRhetosAppOptions.AssetsFolder != null && NonNullBuildOptions.GeneratedAssetsFolder != null && _appOptions.AssetsFolder != _buildOptions.GeneratedAssetsFolder)
+            if (_appOptions?.AssetsFolder != null && _buildOptions?.GeneratedAssetsFolder != null && _appOptions.AssetsFolder != _buildOptions.GeneratedAssetsFolder)

```

Comment:
This: `_appOptions?.AssetsFolder != null && _buildOptions?.GeneratedAssetsFolder != null`
 can be simplified, since we already know at least one is not null? 
Checking only for _appOptions?.AssetsFolder != _buildOptions?.GeneratedAssetsFolder should be equivalent.